Bottom: c98accc6e92fdd92e5f2badf02daa0211e1c7b81
Top:    0efd1a45e76729a602502ee96da409fd24c12650
Author: Ian Kent <ikent@redhat.com>
Date:   2013-04-04 14:53:32 +0800

Refresh of autofs-5.0.7-syncronize-handle_mounts-shutdown.patch

---

diff --git a/daemon/automount.c b/daemon/automount.c
index d2c1ce1..bb5b100 100644
--- a/daemon/automount.c
+++ b/daemon/automount.c
@@ -1348,9 +1348,15 @@ static void *statemachine(void *arg)
 		case SIGINT:
 		case SIGUSR2:
 			master_mutex_lock();
-			if (list_empty(&master_list->mounts)) {
-				master_mutex_unlock();
-				break;
+			if (list_empty(&master_list->completed)) {
+				if (list_empty(&master_list->mounts)) {
+					master_mutex_unlock();
+					return NULL;
+				}
+			} else {
+				if (list_empty(&master_list->mounts)) {
+					master_mutex_unlock();
+					break;
 			}
 			master_mutex_unlock();
